Source/GUI/Predictor/PredictorWidget.cpp

```diff
@@ -25,16 +25,19 @@ void PredictorWidget::initialiseWidgets()
   m_tblStartersPrediction->setSelectionBehavior(QAbstractItemView::SelectRows);
 }
 
-void PredictorWidget::filterUnwanted(const bool filterUnwanted)
+bool PredictorWidget::filterUnwanted(const bool filterUnwanted)
 {
+  bool found = false;
   for (int i = 0; i < m_tblStartersPrediction->rowCount(); i++)
   {
     bool showRow =
         !filterUnwanted || m_tblStartersPrediction->item(i, 2)->background().color().name() ==
                                greenBrush.color().name();
-
+    if (showRow)
+      found = true;
```

Comment:
hum...I don't think this is going to work.  This function is mainly to update the visibility of the rows which is fine for the reroll function because well it has to, but what if you have the hide unwanted UNCHECKED?  `showRow` will always be true because we don't want to hide, but this will always think there is a desired starter which isn'T necessarly true.

Even if that did worked, it's not a good place to put this.  I suggest to split the actuall condition to determine if it's wanted into a separate function (you would pass the index) so filterUnwanted can use it with the OR, but it also allows to use it in a separate new function which would be more suited for this.  Something like `bool predictionHasWantedStarters()`.  This will actually allow to break the loop as soon as one is found too.

Oh also, this had the same problem with the autoReroll function so I guess it would fix that too :)